storage/database/leveldb_database.go

```diff
@@ -415,12 +457,13 @@ hasError:
 }
 
 func (db *levelDB) NewBatch() Batch {
-	return &ldbBatch{db: db.db, b: new(leveldb.Batch)}
+	return &ldbBatch{db: db.db, b: new(leveldb.Batch), ldb: db}
 }
 
 type ldbBatch struct {
 	db   *leveldb.DB
 	b    *leveldb.Batch
+	ldb  *levelDB
```

Comment:
Dosen't this variable `ldb` have the above variables `db` and `b` as well?

Reply:
@jeongkyun-oh `ldb` does have `db`, but not `b`. `db` can be removed if we put `*levelDB` as a member variable. Thanks!